src/common/com/intellij/plugins/haxe/model/type/HaxeTypeCompatible.java

```diff
@@ -141,6 +144,20 @@ else if(from.specifics.length == 0) {
       }
     }
 
+    // check interface/class type overrides
+    HaxeClass baseClass = to.clazz.getHaxeClass();
+    HaxeClass derivedClass = from.clazz.getHaxeClass();
+    if (baseClass != null && derivedClass != null) {
+      List<HaxeType> baseTypes = new ArrayList<HaxeType>();
+      baseTypes.addAll(derivedClass.getHaxeImplementsList());
+      baseTypes.addAll(derivedClass.getHaxeExtendsList());
+      for (HaxeType baseType : baseTypes) {
+        if(to.canAssign(HaxeTypeResolver.getTypeFromType(baseType))) {
```

Comment:
Thinking about this, the algorithm should probably collect all of the inherited classes and put them into a hash, and stop collecting for the node as soon as we find it is already in the hash.  it should probably be added to the HaxeClass class, and not the models (though it can certainly be exposed through the models).